src/utils/confighandler.cpp

```diff
@@ -297,7 +297,11 @@ bool ConfigHandler::verifyLaunchFile()
     bool res = false;
 
 #if defined(Q_OS_LINUX) || defined(Q_OS_UNIX)
-    QString path = QDir::homePath() + "/.config/autostart/Flameshot.desktop";
+    QString path;
+    path = qgetenv("XDG_CONFIG_DIRS") + "/autostart/Flameshot.desktop";
```

Comment:
@hosiet If I understand the request we should actually be checking for the presence of the directory, not the file itself correct?

Reply:
Yes this was not correctly implemented. It should be we check the presence of environment variable to determine whether to use the path from the variable or do the fallback.

And there are many caveats here. What if the directory from envvar does not exist? What if the autostart dir does not exist? They all need to be handled to make the implementation robust.

Reply:
thanks for the feedback,
@hosiet  What shall we do if the `autostart` dir doesn't exist? shall we create it?

Reply:
This is the specification: https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

Reply:
Sorry but these changes do not make sense to me. With `QDir::homePath()` we get directly the $HOME path and this is all we need. 

The handling of the answer from qgetenv() is wrong. The answer is expected to be a single directory but the manual says 
> The directories in $XDG_CONFIG_DIRS should be seperated with a colon ':'.

In my understanding we have to treat the string as a list and see if we get several directories as result and check them one by one.

Last but not least, it makes imho no sense to handle the results of `XDG_CONFIG_DIRS` because the variable is set to /etc/xdg by default and the owner is root. In other words flameshot does not have the privileges to write or change files and the function `void ConfigHandler::setStartupLaunch(const bool start)` will always fail.

Just my two cents, this is just my opinion and should not be a criticism.

Reply:
@kpcee Thanks for chiming in, I agree with the first two points. 

For the 3rd point do you think we should just not follow the XDG Standard?

Reply:
In the source from QDir::homePath() is written that "Under non-Windows operating systems the HOME environment variable is used if it exists, otherwise the path returned by the rootPath()."

If you are concerned that the HOME variable is not set in some distributions or `QDir::homePath()` gives a wrong result or as fallback, I would use  `$XDG_CONFIG_HOME`.

`XDG_CONFIG_DIRS` seems to only list directories from root. This has two disadvantages:
* flameshot has no write permissions and would have to be started as root itself to make changes
* the changes apply system-wide for all users and not only for the currently logged in 

If you want to use the XDG standard instead of `QDir::homePath() `it would have to be done for the current user via `$XDG_CONFIG_HOME/autostart`, it always returns only a directory and no list. This is what I would change in the Pull Request, but I never had problems with `QDir::homePath()`.

Reply:
after reading the reference, seems `XDG_CONFIG_DIRS` is a sort of system-wide config paths, I think `XDG_CONFIG_HOME` is the env that we want to detect.
so I guess
1. check if `XDG_CONFIG_HOME` is set, if it is, use that
2. otherwise, if `XDG_CONFIG_HOME` is not set, we fall back to $HOME/.config, which is the same result of current implementation

PS: not only for autostart file, the config file should follow this rule too ($XDG_CONFIG_HOME/flameshot/flameshot.ini and fall back to ~/.config/flameshot/flameshot.ini) .